src/java/org/apache/cassandra/dht/BootStrapper.java

```diff
@@ -83,6 +86,11 @@ public ListenableFuture<StreamState> bootstrap(StreamStateStore stateStore, bool
             streamer.addRanges(keyspaceName, strategy.getPendingAddressRanges(tokenMetadata, tokens, address));
         }
 
+        if (!initialLocalSchemaVersion.equals(Schema.instance.getVersion()) || !MigrationManager.isReadyForBootstrap())
```

Comment:
i think the order of this check is actually important. `MigrationManager.isReadyForBootstrap()` must be evaluated to true before `initialLocalSchemaVersion.equals(Schema.instance.getVersion())` is evaluated to true. the way it's written now, it's possible for a partially applied schema to complete in between the two checks and not throw here

Reply:
The original idea was to put the second check in the callback for handling a PREPARE, but I decided that that was not necessary.

Moved the checks to SS